Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Instance/Instance.cpp

```diff
@@ -918,5 +925,15 @@ namespace AzToolsFramework
         {
             m_cachedInstanceDom.CopyFrom(instanceDom->get(), m_cachedInstanceDom.GetAllocator());
         }
+
+        bool Instance::IsCachedInstanceDomUpToDate() const
```

Comment:
This looks more like a helper/utils function than a member function. I recommend that you move this to PrefabTestFixture class. Also, 'UpToDate' can be misleading when there are overrides. To be more clear, you can change it to something like DoesCachedInstanceDomMatchTemplateDom